realsense2_camera/test/live_camera/rosci.py

```diff
@@ -0,0 +1,144 @@
+            if stdout is None:
+                sys.stdout.flush()
+            elif stdout and stdout != subprocess.PIPE:
+                if append:
+                    handle = open( stdout, "a" )
+                    handle.write(
+                        "\n----------TEST-SEPARATOR----------\n\n" )
+                    handle.flush()
+                else:
+                    handle = open( stdout, "w" )
+
+            result = subprocess.run( cmd,
+                    stdout=handle,
+                    stderr=subprocess.STDOUT,
+                    universal_newlines=True,
+                    timeout=200,
+                    check=True )
+
+        except Exception as e:
+             log.w( "Error Exception:\n ",e )
+                         
+        finally:
+            if handle:
+                handle.close()      
+
+device_set = None
+try:
+    opts, args = getopt.getopt( sys.argv[1:], 'hvqr:st:',
+                                longopts=['help', 'verbose', 'debug', 'quiet', 'regex=', 'stdout', 'tag=', 'list-tags',
+                                          'list-tests', 'no-exceptions', 'context=', 'repeat=', 'config=', 'no-reset', 'hub-reset',
+                                          'rslog', 'skip-disconnected', 'live', 'not-live', 'device='] )
+except getopt.GetoptError as err:
+    log.e( err )  # something like "option -a not recognized"
+    usage()
+for opt, arg in opts:
+    if opt in ('-h', '--help'):
+        usage()
+    elif opt == '--device':
+        device_set = arg.split()
+        for dev in device_set:
+             print(dev)        
+#Get into action    
+try:
+    logdir = os.path.join( os.path.dirname( os.path.dirname( os.path.dirname( os.path.abspath( __file__ )))), 'log')
+    os.makedirs( logdir, exist_ok=True )
+    print(logdir)
+    
+   
+
+    #Import '_device_by_sn' from devices.py module of librealsense repo
+    from rspy import devices
+    if not devices.hub:
+         assert False, 'hub not available'
+    else:
+        devices.hub.connect()
+    devices.query( hub_reset = hub_reset )
+    #devices.map_unknown_ports()
+    global _device_by_sn         
+
+    if not devices._device_by_sn:
+         assert False, 'No Camera Devices detected!'
+    else:
+         #Loop in for all devices and run tests
+         for device in devices._device_by_sn.values():
+              print(f'\033[93m Device found: {device.name} \033[0]')
+         for device in devices._device_by_sn.values():
+            if device.name == 'D455':
+                print('Running test on device:', device.name)
+                cmd = command(str(device.name).lower())
+                run_test(cmd, device.name, stdout=logdir, append =False)
+            elif device.name != 'D455':
+                 print('Skipping test on device:', device.name)
+
+finally:
+        devices.hub.disconnect()
+        run_time = time.time() - start_time
+        log.d( "server took", run_time, "seconds" )
+
+sys.exit( 0 )
```

Comment:
Will the current script return ~= 0 if the test failed and 0 if it pass?
Can you add a screen shot of the running output?

Thanks

Reply:
![image](https://github.com/IntelRealSense/realsense-ros/assets/158188917/be92f76e-3bf1-4396-9595-cc056e97552c)


![image](https://github.com/IntelRealSense/realsense-ros/assets/158188917/ff3aa2d3-7196-41a4-908f-f4dae4ebfd75)

Reply:
Yes, that's right. Made it little more clear in commit #3.

Reply:
**Test ROS** stage fails when the **subprocess.run()**  returns non zero exit status. However, there is NO log processing mechanism implemented yet as to why the test failed. For now, logs have to be checked manually if test fails @ "Build artifacts": ros2/realsense2_camera/log/